### DISCUSSION TITLE
How to groupBy relationship's column with descending order

### DISCUSSION THREAD

**Discussion by myckhel** [laravel/framework#35782] [Q&A]
Given the example below
the latest order has no effect on the query
its grouped by the first item instead of the last item in the table

```php
function events(){
  return $this->hasMany(ChatEvent::class, 'made_id')
  ->where('made_type', 'App\\Models\\Conversation')
  ->groupBy('type')->latest();
}
```

|  id | type  |  created_at
|---|---|---|
| 1  |  deliver | 2020-12-27 12:11:05
|  2 | deliver  | 2020-12-28 12:11:05
| 3  |  deliver | 2021-01-05 09:17:20 | 2021-01-05 09:17:20

sql output
```sql
select * from `chat_events` where `made_type` = ? group by `type` order by `created_at` desc
```

actual result
```json
"chat_events": {
      "deliver": {
          "id": 1,
          "type": "deliver",
          "created_at": "2020-12-27T12:11:05.000000Z"
      },
  },
```

expected result
```json
"chat_events": {
      "deliver": {
          "id": 3,
          "type": "deliver",
          "created_at": "2021-01-05T09:17:20.000000Z"
      },
  },
```

Please how can i make it retrieve expected result? @taylorotwell @themsaid @driesvints @lucasmichot

**Comment by lucasmichot** [comment]
https://stackoverflow.com/questions/31857961/order-by-before-group-by-using-eloquent-laravel

**Reply by myckhel** [reply]
Weeeew.
thanks but seems difficult todo with relationship.

i wonder why sql grouping so complicated.

**Comment by driesvints** [comment]
Please don't mass mention people. Someone will help you when they see this.

**Reply by myckhel** [reply]
Okay
cool.
